modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/MuleArtifactClassLoader.java

```diff
@@ -109,21 +126,57 @@ public void setResourceReleaserClassLocation(String resourceReleaserClassLocatio
     this.resourceReleaserClassLocation = resourceReleaserClassLocation;
   }
 
-  protected ResourceReleaser createResourceReleaserInstance() {
+  public void setErrorHooksClassLocation(String errorHooksClassLocation) {
+    this.errorHooksClassLocation = errorHooksClassLocation;
+  }
+
+  protected Object createCustomInstance(String classLocation) {
     InputStream classStream = null;
     try {
-      classStream = this.getClass().getResourceAsStream(resourceReleaserClassLocation);
+      classStream = this.getClass().getResourceAsStream(classLocation);
       byte[] classBytes = IOUtils.toByteArray(classStream);
       classStream.close();
       Class clazz = this.defineClass(null, classBytes, 0, classBytes.length);
-      return (ResourceReleaser) clazz.newInstance();
+      return clazz.newInstance();
     } catch (Exception e) {
-      throw new RuntimeException("Can not create resource releaser instance from resource: " + resourceReleaserClassLocation, e);
+      throw new RuntimeException("Can not create instance from resource: " + classLocation, e);
     } finally {
       closeQuietly(classStream);
     }
   }
 
+  protected ResourceReleaser createResourceReleaserInstance() {
+    return (ResourceReleaser) createCustomInstance(resourceReleaserClassLocation);
+  }
+
+  /**
+   * Setup reactor-core error hooks, these are required for plugins that end up executing reactive streams.
+   *
+   * For instance, compatibility plugin which inherits from transformers that call {@code block()}.
+   */
+  protected void configureErrorHooks() {
```

Comment:
To be called in child constructors when initialization requires another steps before.